iceoryx2-bb/container/src/queue.rs

```diff
@@ -452,13 +456,13 @@ pub mod details {
 /// Relocatable queue with compile time fixed size capacity. In contrast to its counterpart the
 /// [`Queue`] it is movable.
 #[repr(C)]
-#[derive(Debug)]
-pub struct FixedSizeQueue<T, const CAPACITY: usize> {
+#[derive(Debug, ZeroCopySend)]
```

Comment:
No we cannot use the derive macro here. It is completely fine that the user uses the `FixedSizeQueue` with `String` for instance which does not implement `ZeroCopySend`.

It must be implemented by hand for every type which is `T: ZeroCopySend` like you did in case of the `MetaQueue`.